src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -98,10 +118,10 @@ public void replaceEntityTypeDefinitions(List<EntityType> entityTypes) {
     });
   }
 
-  private List<EntityTypeColumn> fetchColumnNamesForCustomFields(UUID entityTypeId, EntityType entityType) {
+  private List<EntityTypeColumn> fetchColumnNamesForCustomFields(String entityTypeId, EntityType entityType, Map<String, EntityType> rawEntityTypes) {
     log.info("Getting columns for entity type ID: {}", entityTypeId);
-    EntityType entityTypeDefinition = entityTypeId.toString().equals(entityType.getId()) ? entityType :
-      getEntityTypeDefinition(entityTypeId, null).orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    EntityType entityTypeDefinition = entityTypeId.equals(entityType.getId()) ? entityType :
+      Optional.ofNullable(rawEntityTypes.get(entityTypeId)).orElseThrow(() -> new EntityTypeNotFoundException(UUID.fromString(entityTypeId)));
```

Comment:
I had to change this, since the previous approach would make it try to load from the cache while it's still trying to populate the cache, which causes an exception